src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -961,12 +961,38 @@ public override BoundNode VisitPassByCopy(BoundPassByCopy node)
 
         public override BoundNode VisitIsPatternExpression(BoundIsPatternExpression node)
         {
-            Debug.Assert(!IsConditionalState);
-            VisitRvalue(node.Expression);
-
             bool negated = node.Pattern.IsNegated(out var pattern);
             Debug.Assert(negated == node.IsNegated);
 
+            Debug.Assert(!IsConditionalState);
+            if (VisitPossibleConditionalAccess(node.Expression, out var stateWhenNotNull))
+            {
+                switch (GetTopLevelNullTest(pattern))
```

Comment:
>GetTopLevelNullTest [](start = 24, length = 19)

It looks like this is only used here. Consider making it a local function.

Reply:
Done